src/components/AppSelector/AppSelector.tsx

```diff
@@ -13,59 +14,105 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
-import { useRef } from 'react'
+import { useEffect, useRef, useMemo, useState } from 'react'
+import { SelectInstance } from 'react-select'
 import {
-    abortPreviousRequests,
-    APP_SELECTOR_STYLES,
-    AppSelectorDropdownIndicator,
-    AppSelectorNoOptionsMessage,
+    SelectPickerVariantType,
+    ComponentSizeType,
+    useAsync,
+    SelectPicker,
+    SelectPickerProps,
 } from '@devtron-labs/devtron-fe-common-lib'
-import { Props as SelectProps, SelectInstance } from 'react-select'
-import AsyncSelect from 'react-select/async'
-import { appListOptions } from './AppSelectorUtil'
-
-interface AppSelectorType {
-    onChange: ({ label, value }) => void
-    appId: number
-    appName: string
-    isJobView?: boolean
-}
+import { AppSelectorType, RecentlyVisitedSelectPickerTypes } from './types'
+import { getDropdownOptions, fetchRecentlyVisitedDevtronApps, getNoOptionsMessage } from './AppSelectorUtil'
+import { fetchAllAppListGroupedOptions } from './AppSelectorService'
 
-const AppSelector = ({ onChange, appId, appName, isJobView }: AppSelectorType) => {
+const AppSelector = ({
+    onChange,
+    appId,
+    appName,
+    isJobView,
+    recentlyVisitedDevtronApps,
+    setRecentlyVisitedDevtronApps,
+}: AppSelectorType) => {
     const selectRef = useRef<SelectInstance>(null)
-
     const abortControllerRef = useRef<AbortController>(new AbortController())
+    const [options, setOptions] = useState([])
+    const [inputValue, setInputValue] = useState('')
+    const [areOptionsLoading, setAreOptionsLoading] = useState(false)
+
+    const [, result] = useAsync(
+        () => fetchRecentlyVisitedDevtronApps(appId, appName),
+        [appId, appName],
+        !!appName && !!appId,
+    )
+
+    useEffect(() => {
+        if (result) {
+            setRecentlyVisitedDevtronApps(result)
+        }
+    }, [result])
+
+    const memoizedDropdownOptions = useMemo(
+        () => () => getDropdownOptions(inputValue, recentlyVisitedDevtronApps, appId),
```

Comment:
no need for function 


```suggestion
        () => getDropdownOptions(inputValue, recentlyVisitedDevtronApps, appId),
```